src/pkg/services/m365/m365.go

```diff
@@ -29,9 +29,13 @@ type ServiceAccess struct {
 
 // User is the minimal information required to identify and display a user.
 type User struct {
-	PrincipalName string
-	ID            string
-	Name          string
+	PrincipalName      string
+	ID                 string
+	Name               string
+	UserPurpose        string
+	HasMailBox         bool
+	HasOnedrive        bool
```

Comment:
This overlaps with UserInfo, which seems like the better owner for this sort of metadata.

Reply:
To be clear, we have
- a struct named `UserInfo`
- a func named `UserInfo`

Do we want to accommodate the implementation in the `UserInfo` function and these value go as part of `UserInfo` struct?

Reply:
As of now I have merged the implementation and the struct too. The only issue I see with this is-
In cases where only `HasMailBox` info was required, we make too more calls. Which might be a overhead.